src/js/player.js

```diff
@@ -2747,22 +2803,53 @@ class Player extends Component {
   }
 
   /**
-   * Get or set the autoplay attribute.
+   * Get or set the autoplay option. When this is a boolean it will
+   * modify the attribute on the tech. When this is a string the attribute on
+   * the tech will be removed and `Player` will handle autoplay on loadstarts.
    *
-   * @param {boolean} [value]
-   *        - true means that we should autoplay
-   *        - false means that we should not autoplay
+   * @param {boolean|string} [value]
+   *        - true: autoplay using the browser behavior
+   *        - false: do not autoplay
+   *        - 'play': call play() on every loadstart
+   *        - 'muted': call muted() then play() on every loadstart
+   *        - 'any': call play() on every loadstart. if that fails call muted() then play().
+   *        - *: values other than those listed here will be set `autoplay` to true
    *
-   * @return {string}
+   * @return {boolean|string}
    *         The current value of autoplay when getting
    */
   autoplay(value) {
-    if (value !== undefined) {
-      this.techCall_('setAutoplay', value);
+    // getter usage
+    if (value === undefined) {
+      return this.options_.autoplay || false;
+    }
+
+    let techAutoplay;
+
+    // if the value is a valid string set it to that
+    if (typeof value === 'string' && (/(any|play|muted)/).test(value)) {
```

Comment:
should we call `this.manualAutoplay_` in here? Use case would be something like:
```js
var player = videojs('vid');

player.autoplay('any');
```
I'd expect the player to autoplay in this case. This would also match current behavior (in browsers that support autoplay). Main risk is that we'd want to be able to do something like the following too
```js
var player = videojs('vid', { autoplay: 'any' );

player.autoplay(false);
```
and have it not autoplay.

Also, should we have a string state "off" and deprecate boolean values for it?

Reply:
Maybe just do this in a followup PR?

Reply:
Should `player.autoplay('any')` also return a promise?

Reply:
I think we probably just want events rather than a promise